zerver/outgoing_webhooks/slack.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import absolute_import
+from typing import Any, Dict, Text, Tuple, Callable, Mapping
+import json
+from requests import Response
+from zerver.models import UserProfile, email_to_domain
+
+from zerver.lib.outgoing_webhook import OutgoingWebhookServiceInterface
+
+class SlackOutgoingWebhookService(OutgoingWebhookServiceInterface):
+
+    def __init__(self, base_url, token, user_profile, service_name):
+        # type: (Text, Text, UserProfile, Text) -> None
+        super(SlackOutgoingWebhookService, self).__init__(base_url, token, user_profile, service_name)
+
+    def process_event(self, event):
+        # type: (Dict[str, Any]) -> Tuple[Dict[str ,Any], Dict[str, Any]]
+        rest_operation = {'method': 'POST',
+                          'relative_url_path': '',
+                          'base_url': event['base_url'],
+                          'request_kwargs': {}}
+
+        if event['message']['type'] == 'private':
+            raise NotImplementedError("Private messaging service not supported.")
+
+        request_data = {"token": event['token'],
```

Comment:
How did you figure out the list of keys necessary in the `request_data`?  Did you look at some documentation?  If so, can you add the link in a comment?